File: custom-modules/aws/src/main/webapp/js/license.js
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

$(document).ready(function () {
    var $licenseForm = $("#licenseForm");
    var $error = $("#error");
    $( "#datepicker" ).datepicker();

//    $("#installBtn").click(function () {
//        $error.hide();
//        $licenseForm.submit();
//    });


    $licenseForm.submit(function (e) {
        e.preventDefault();
        $error.hide();
        var $datepicker = $("#datepicker");
        var $date = $datepicker.val();
        var $milliseconds  = Date.parse($date);
        $datepicker.val($milliseconds);

        $.ajax({
            url: "license/download",
            method: 'POST',
            dataType: 'json',
            data: JSON.stringify($("#licenseForm").serializeObject()),
            contentType: 'application/json',
            type: 'application/json',

            success: function (data) {
                if (data.keyFileName) {
                    var obj = { file: data.keyFileName };
                    window.location.href = 'install/downloadKey?' + $.param(obj);

                    $licenseForm.hide();
                    $("#installBtn").hide();
                    $("#successMsg").html(data.status);
                    $("#successMsg").show();
                } else {
                    $error.html(data.status);
                    $error.show();
                }
                $datepicker.val($date);
            },
            error: function (data, status, er) {
                $error.text(data.responseJSON.error).show();
            }
        });
    });

});

$(document).ajaxStart(function() {
    $("body").addClass("loading");
});
$(document).ajaxStop(function() {
    $("body").removeClass("loading");
});

$.fn.serializeObject = function () {
    var o = {};
    var a = this.serializeArray();
    $.each(a, function () {
        if (o[this.name]) {
            if (!o[this.name].push) {
                o[this.name] = [o[this.name]];
            }
            o[this.name].push(this.value || '');
        } else {
            o[this.name] = this.value || '';
        }
    });
    return o;
};

File: custom-modules/aws/src/main/java/net/firejack/aws/web/model/License.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package net.firejack.aws.web.model;


import org.springframework.beans.factory.config.BeanDefinition;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlRootElement;
import java.io.Serializable;

@Component
@XmlRootElement
@Scope(BeanDefinition.SCOPE_PROTOTYPE)
@XmlAccessorType(XmlAccessType.FIELD)
public class License implements Serializable {
    private static final long serialVersionUID = -5885049331615350740L;

    private String name;
    private Integer session;
    private Long expired;
    private String signature;
    private String accessPassword;

    public License() {
    }

    public License(String name, Integer session, Long expired) {
        this.name = name;
        this.session = session;
        this.expired = expired;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getSession() {
        return session;
    }

    public void setSession(Integer session) {
        this.session = session;
    }

    public Long getExpired() {
        return expired;
    }

    public void setExpired(Long expired) {
        this.expired = expired;
    }

    public String getSignature() {
        return signature;
    }

    public void setSignature(String signature) {
        this.signature = signature;
    }

    public String getAccessPassword() {
        return accessPassword;
    }

    public void setAccessPassword(String accessPassword) {
        this.accessPassword = accessPassword;
    }
}
